### PR TITLE
Implement support for tailoring variables

### PR DESCRIPTION
**Pull request by jhrozek**
Implements a new CR called Variable that implements a subset of the
XCCDF standard related to setting variables during tailoring and
rendering them to a tailoring file using the "set-value" tailoring
element. The subset supported roughly equals to variable types and a
selection of allowed values.

The variables are parsed along with the type and selections in the
profileparser pod. Note that the variable parsing is implemented in a
new profileparser package that is just called from the main package of
the profileparser pod. A future patch should move the rest of the main
package there also to make it testable.

See
`deploy/crds/compliance.openshift.io_v1alpha1_tailoredprofile_cr.yaml`
for an example on how to tune a variable. The tailoredprofile controller
looks up the referenced variables and creates a slice of new Variable
instances passed to the tailoring engine which finally renders the
tailoring XML.

Note that the validation of variable types and selections is done in the
tailoredprofile when looking up the variable and matching it with the
TP's spec: this is done so that we can surface the errors to the TP
status and at the same time have the full information in the Variable CR
available in tailoring if needed.

Closes: #3

### CODE REVIEW

**Comment by jhrozek** [comment]
There's still some follow-ups needed, notably e2e tests, reducing code duplication and maybe support for a wider subset of the XCCDF standard if needed (e.g. the OCP DS already uses a regex-checker for one of the variables..), but let's start the review process already..

**Comment by JAORMX** [comment]
Not a lot of comments on my side. Spot on testing. I like moving the codebase for the profile parser to its own package. It's good stuff!

File: pkg/apis/compliance/v1alpha1/variable_types.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// FIXME: several values are shared with Rule object, maybe create a shared
+	// struct? The shared values are documented in table 5 of the XCCDF spec
+	// The XCCDF ID
+
+	// the ID of the variable
+	ID string `json:"id"`
+	// The title of the Rule
```
[the remark this answers is not in the document]
Comment (jhrozek) [reply pkg/apis/compliance/v1alpha1/variable_types.go]: Fixed here and elsewhere

File: pkg/apis/compliance/v1alpha1/tailoredprofile_types.go [hunk 2]
```diff
@@ -4,6 +4,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// FIXME: move name/rationale to a common struct with an interface?
+
```
[the remark this answers is not in the document]
Comment (jhrozek) [reply pkg/apis/compliance/v1alpha1/tailoredprofile_types.go:8]: OK, I didn't want to also refactor code without having an agreement on how the general idea works at the moment, but I'll do this in a subsequent patch.

File: pkg/apis/compliance/v1alpha1/variable_types.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+// Variable describes a tunable in the XCCDF profile
+// +kubebuilder:resource:path=variables,scope=Namespaced
+type Variable struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// FIXME: several values are shared with Rule object, maybe create a shared
+	// struct? The shared values are documented in table 5 of the XCCDF spec
+	// The XCCDF ID
```
[the remark this answers is not in the document]
Comment (jhrozek) [reply pkg/apis/compliance/v1alpha1/variable_types.go:36]: Same as above, I think this makes sense in a follow up

File: pkg/profileparser/profileparser_test.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		It("Has the expected description", func() {
+			const expectedDescription = "Specify whether and how sshd separates privileges when handling incoming network connections."
+			Expect(sshdPrivSepVar.Description).To(BeEquivalentTo(expectedDescription))
+		})
+
+		It("Has the expected selections", func() {
+			Expect(sshdPrivSepVar.Selections).To(ConsistOf([]cmpv1alpha1.ValueSelection{
+				{
+					"no",
+					"no",
```
Comment (JAORMX) [inline pkg/profileparser/profileparser_test.go:91]: could you add a key to these? `make verify` complains about it.

**Pull request merged**